### PR TITLE
Remove some print statements within algorithmic functions

### PR DESCRIPTION
**Pull request by tianyizheng02**
### Describe your change:

Removed some print statements within algorithmic functions (i.e., code that's not within a `__main__`, a`main` function, or a utility function).

Encapsulated some test/demo code that wasn't within `__main__`s.

This PR contributes to #7337 but is not a complete fix.

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by tianyizheng02** [comment]
Somehow the build is failing with `Fatal Python error: Illegal instruction` when running pytest... @CaedenPH @cclauss any idea why this could be happening?

File: dynamic_programming/sum_of_subset.py [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
     False
-    >>> is_sum_subset([2, 4, 6, 8], 4, 14)
+    >>> is_sum_subset([2, 4, 6, 8], 14)
     True
     """
     # a subset value says 1 if that subset sum can be formed else 0
     # initially no subsets can be formed hence False/0
-    subset = [[False for i in range(required_sum + 1)] for i in range(arr_len + 1)]
+    arr_len = len(arr)
+    subset = [[False for _ in range(required_sum + 1)] for _ in range(arr_len + 1)]
```
Comment (cclauss) [inline dynamic_programming/sum_of_subset.py:11]:
```suggestion
    subset = [[False] * (required_sum + 1) for _ in range(arr_len + 1)]
```

File: machine_learning/forecasting/run.py [hunk 2]
```diff
@@ -107,50 +107,55 @@ def data_safety_checker(list_vote: list, actual_result: float) -> None:
...
+    tst_user = total_user[len(total_user) - 1 :]
+    tst_match = total_match[len(total_match) - 1 :]
+
+    # voting system with forecasting
+    res_vote = []
+    res_vote.append(
+        linear_regression_prediction(trn_date, trn_user, trn_match, tst_date, tst_match)
+    )
+    res_vote.append(sarimax_predictor(trn_user, trn_match, tst_match))
+    res_vote.append(support_vector_regressor(x_train, x_test, trn_user))
```
Comment (cclauss) [inline machine_learning/forecasting/run.py:155]:
```suggestion
    res_vote = [
        linear_regression_prediction(trn_date, trn_user, trn_match, tst_date, tst_match),
        sarimax_predictor(trn_user, trn_match, tst_match),
        support_vector_regressor(x_train, x_test, trn_user),
    ]
```

File: machine_learning/forecasting/run.py [hunk 3]
```diff
@@ -107,50 +107,55 @@ def data_safety_checker(list_vote: list, actual_result: float) -> None:
...
+
+    # check the safety of today's data
+    if data_safety_checker(res_vote, tst_user):
+        print("Today's data is safe.")
+    else:
+        print("Today's data is not safe.")
```
Comment (cclauss) [inline machine_learning/forecasting/run.py:161]:
```suggestion
    not_str = "" if data_safety_checker(res_vote, tst_user) else "not "
    print("Today's data is {not_str}safe.")
```

**Review by cclauss — APPROVED** [review]

File: machine_learning/forecasting/run.py [hunk 4]
```diff
@@ -107,50 +107,54 @@ def data_safety_checker(list_vote: list, actual_result: float) -> None:
...
+        linear_regression_prediction(
+            trn_date, trn_user, trn_match, tst_date, tst_match
+        ),
+        sarimax_predictor(trn_user, trn_match, tst_match),
+        support_vector_regressor(x_train, x_test, trn_user),
+    ]
+
+    # check the safety of today's data
+    not_str = "" if data_safety_checker(res_vote, tst_user) else "not "
+    print("Today's data is {not_str}safe.")
```
Comment (tianyizheng02) [inline machine_learning/forecasting/run.py:160]: This string needs to be an f-string for `not_str` to print correctly

**Pull request merged**